src/components/list/LibraryList.vue

```diff
@@ -125,6 +142,64 @@ function onItemClick(id) {
     selectedItemId.value = null;
   }
 }
+
+/**
+ * Load templates.
+ * @param {number} page - Current page for pagination.
+ * @returns {Promise<void>} Promise with nothing on success otherwise an error.
+ */
+async function loadTemplates(page) {
+  loading.value = true;
+  currentPage.value = page;
+  const name = definitionFilter.value ? definitionFilter.value.trim() : '';
+
+  return getTemplatesByType({
+    HAS_BACKEND: process.env.HAS_BACKEND,
+    TEMPLATE_LIBRARY_BASE_URL: process.env.TEMPLATE_LIBRARY_BASE_URL,
+  }, 'COMPONENT', {
+    plugin: props.plugin.data.name,
+    count: 15,
+    page: currentPage.value,
+    name,
+  }).then((data) => {
+    haveMoreTemplates.value = data.pageable.pageNumber + 1 < data.totalPages;
+    currentPage.value = data.pageable.pageNumber;
+    totalTemplates.value = data.totalElements;
+
+    return data.content.map((template) => ({
+      ...template,
+      isTemplate: true,
+    }));
+  }).finally(() => {
+    loading.value = false;
+  });
+}
+
+/**
+ * Initialize templates by loading page 0.
+ * @returns {Promise<void>} Promise with nothing on success otherwise an error.
+ */
+async function initTemplates() {
+  return loadTemplates(0).then((data) => {
+    templates.value = data;
+  });
+}
+
+/**
+ * Load templates for next page after the current.
+ * @returns {Promise<void>} Promise with nothing on success otherwise an error.
+ */
+async function loadMoreTemplates() {
+  return loadTemplates(currentPage.value + 1).then((data) => {
```

Comment:
I don't really like the currentPage.value + 1 in the method. Its less readable and it cost nothing to have another line for that.
But I guess its a point of view